Code/Source/Lidar/LidarRaycaster.cpp

```diff
@@ -234,33 +229,50 @@ namespace RGL
+
         // We need to configure if points should be compacted to minimize the CPU operations when retrieving raycast results.
-        ConfigurePointsCompact(
-            !((m_resultFlags & ROS2::RaycastResultFlags::Ranges) == ROS2::RaycastResultFlags::Ranges) && !m_addMaxRangePoints);
+        m_graph.SetIsCompactEnabled(ShouldEnableCompact());
+
+        m_graph.SetIsPublishingEnabled(ShouldEnablePublishing());
     }
 
-    void LidarRaycaster::ConfigurePointsCompact(bool value)
+    void LidarRaycaster::ConfigurePointCloudPublisher(
+        const AZStd::string& topicName, const AZStd::string& frameId, const ROS2::QoS& qoSPolicy)
     {
-        if (m_useCompact == value) // Make sure to only allow state switching calls
-        {
-            return;
-        }
+        RGL_CHECK(rgl_node_points_transform(&m_graph.GetNodes().m_pointCloudTransform, &Utils::IdentityTransform));
+
+        RGL_CHECK(rgl_node_points_ros2_publish_with_qos(
+            &m_graph.GetNodes().m_pointCloudPublish,
+            topicName.c_str(),
+            frameId.c_str(),
+            static_cast<rgl_qos_policy_reliability_t>(static_cast<int>(qoSPolicy.GetQoS().reliability())),
+            static_cast<rgl_qos_policy_durability_t>(static_cast<int>(qoSPolicy.GetQoS().durability())),
+            static_cast<rgl_qos_policy_history_t>(static_cast<int>(qoSPolicy.GetQoS().history())),
+            qoSPolicy.GetQoS().depth()));
+
+        m_graph.SetIsPublishingEnabled(ShouldEnablePublishing());
+    }
 
-        m_useCompact = value;
+    bool LidarRaycaster::CanHandlePublishing()
+    {
+        return m_graph.IsPublishingEnabled();
+    }
 
-        // We unpin / repin the compact node from / to the rgl tree.
-        if (value)
-        {
-            Utils::ErrorCheck(rgl_graph_node_remove_child(m_rayTraceNode, m_pointsFormatNode));
-            Utils::ErrorCheck(rgl_graph_node_add_child(m_rayTraceNode, m_pointsCompactNode));
-            Utils::ErrorCheck(rgl_graph_node_add_child(m_pointsCompactNode, m_pointsFormatNode));
-        }
-        else
-        {
-            Utils::ErrorCheck(rgl_graph_node_remove_child(m_rayTraceNode, m_pointsCompactNode));
-            Utils::ErrorCheck(rgl_graph_node_remove_child(m_pointsCompactNode, m_pointsFormatNode));
-            Utils::ErrorCheck(rgl_graph_node_add_child(m_rayTraceNode, m_pointsFormatNode));
-        }
+    void LidarRaycaster::UpdatePublisherTimestamp(AZ::u64 timestampNanoseconds)
+    {
+        RGL_CHECK(rgl_scene_set_time(nullptr, timestampNanoseconds));
+    }
+
+    bool LidarRaycaster::ShouldEnableCompact() const
+    {
+        const bool RangesExpected = (m_resultFlags & ROS2::RaycastResultFlags::Ranges) == ROS2::RaycastResultFlags::Ranges;
+        return !RangesExpected && !m_isMaxRangePoints;
+    }
+
+    bool LidarRaycaster::ShouldEnablePublishing() const
+    {
+        // Same conditions.
+        return m_graph.IsPublisherConfigured() && ShouldEnableCompact();
```

Comment:
It could be cleaner if you would check for PointsExpected.
`PointsExpected && !m_isMaxRangePoints;`

Also, it would be nice to describe, why these conditions must be satisfied (in both functions: `ShouldEnableCompact` and `ShouldEnablePublishing`).